### ISSUE TITLE
Handlebars dynamic functionality

### ISSUE THREAD

**Issue by bkurowski** [samtecspg/articulate#441]
### Question

I am trying to work with a dynamic function just like below:

`The id of {{slots.Cells.original}} equals {{#forEach webhookResponse}}{{#contains name slots.Cells.original}}{{id}} {{/contains}}{{/forEach}}`
ANSWER: Sorry can you rephrase that ? 

So what happens here is that when I try to complete the function just like above it does not pass the Slot value inside `{{#Contains}}` function. 
When I tried not dynamic (hardcoded) it has worked no problem.

`The id of {{slots.Cells.original}} equals {{#forEach webhookResponse}}{{#contains name "node1"}}{{id}} {{/contains}}{{/forEach}}`
ANSWER: The id of node1 equals 1234567hhgdt33234

Not sure what I am doing wrong in this query. Can anyone see any issue ? 
Might be a stupid question but does `{{#contains}}` allow usage of slots inside of it ? 

Thank you,
Kind Regards

**Comment by bkurowski** [comment]
Hi @wrathagom. I know that fallback has nothing to do with handlebars. That just gives me indication that my handlebar function placed in agent response did not get resolved as expected.

I understand what do you mean with threshold, I did not change it from start it is always set up at 50%. 
I have established 60% or 70% or even lower then 50% , 20 or 30 %. That did not change the way it is acting.
It still does not pass the variable to the handlebar function. While hard coded version works with no problem.
I can not really see the reason of this not to work.
Any ideas ? 

Thank you for your time,

Kind Regards,
Bart

**Comment by bkurowski** [comment]
I have observed that `{{#forEach}}` handlebar causes this issue. When i get rid of `{{#forEach}}` statement, then it is passing the variable with no problems to my agent response. The thing is i need forEach in my agent response as it is iterating through my JSON endpoint do find "Several keys named "name".

I am just saying what i have observed. But is there any way to bypass that issue ? 

Kind Regards,
Thank you,
Bart

**Comment by wrathagom** [comment]
@bkurowski if I am not mistaken the problem is that once inside the `{{#forEach}}` you are no longer accessing the top level object, but rather each object inside the loop.

You need to go up an object, similar to how you would go up a directory.

```
The id of {{slots.Cells.original}} equals {{#forEach webhookResponse}}{{#contains name ../slots.Cells.original}}{{id}} {{/contains}}{{/forEach}}
```

notice the `../` before slots. Try that and let me know.

**Comment by bkurowski** [comment]
Thank you ! that was it. Sorry for such questions, I just could not figure it out. 
Thanks !

Best Regards,
Bart
